### PR TITLE
feat: add refactoring suggestions docs

### PR DESCRIPTION
**Pull request by philippfromme**
Adds documentation for the new refactoring suggestions feature that is part of the 8.5 release.

Related to https://github.com/camunda/product-hub/issues/2010

image

## Description

Adds documentation for the new refactoring suggestions alpha feature.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by christinaausley** [comment]
Let me know when you're ready for review @philippfromme 👍

**Comment by philippfromme** [comment]
We've integrated a new version. I will update the screenshots.

**Comment by philippfromme** [comment]
Ready for review.

**Comment by lmbateman** [comment]
A couple of comments/questions:

1. The Note at the beginning mentions the alpha features menu. If I understand correctly, the alpha features menu has been enabled for all users, but because that was done recently, trial/free users may not know anything about it. Would it be possible to add either a link to the page where the menu is documented, or a brief description of where to find it? (E.g., "...through the alpha features menu in the Console Organization Settings," or something like that.)
2. To help prevent user frustration, I think we should add that currently the feature is only available for named blank elements; that not every named blank element will have suggestions; and perhaps that you might be able to get different suggestions by editing the name of your element. Maybe something along these lines?
"Since this is an alpha feature, only blank elements with names generate suggestions, and suggestions are not available for every named blank element. If there are suggestions available, you will see a list of them. Click on a suggestion to apply it to your diagram: ... 
If there are no suggestions available, you will see a message indicating that: ...
You can experiment with generating suggestions by editing the name of the element. Try:
- Adding or subtracting words -- for example, changing “Create issue” to “Create GitHub issue”
- Changing words to synonyms -- for example, “message” to “email”

File: docs/components/modeler/web-modeler/advanced-modeling/img/refactoring-suggestions-3.png
Comment (christinaausley) [inline docs/components/modeler/web-modeler/advanced-modeling/img/refactoring-suggestions-3.png:1]: I think this should read "There are currently no **suggestions** available for this element" -- can we make an adjustment to the UI? This isn't blocking, but wanted to call it out regardless.

Comment (philippfromme) [reply docs/components/modeler/web-modeler/advanced-modeling/img/refactoring-suggestions-3.png:1]: Ooops. We'll have to fix that in the code. 🙈

Comment (philippfromme) [reply docs/components/modeler/web-modeler/advanced-modeling/img/refactoring-suggestions-3.png:1]: I just fixed it in the screenshots for now.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**